### PR TITLE
kodi: update to 17.0-alpha3-fc46cf2

### PR DESCRIPTION
**Pull request by MilhouseVH**
Tweak `tools/mkpkg/mkpkg_kodi` so that a (relatively) recent history is cloned, in order to package a specific rev (mirror already updated).

Updated `tools/RPi/rpi-kodi-rebase.sh` to exclude (drop) the "UNSTABLE:" placeholder.

Bump ffmpeg to 3.1.1, keeping in sync with Kodi 17a3.

### CODE REVIEW

File: packages/mediacenter/kodi-theme-Estuary/package.mk [hunk 1]
```diff
@@ -50,7 +50,6 @@ makeinstall_target() {
     cp -R */ $INSTALL/usr/share/kodi/addons/skin.estuary
     cp *.txt $INSTALL/usr/share/kodi/addons/skin.estuary
     cp *.xml $INSTALL/usr/share/kodi/addons/skin.estuary
-    cp *.png $INSTALL/usr/share/kodi/addons/skin.estuary
```
Comment (lrusak) [inline packages/mediacenter/kodi-theme-Estuary/package.mk]: is this just a leftover?

File: packages/mediacenter/kodi/scripts/kodi-config [hunk 2]
```diff
@@ -42,13 +42,6 @@ if [ ! -f $HOME/.kodi/userdata/guisettings.xml ] ; then
   fi
 fi
 
-# cleanup *.fi cache files from .kodi/temp as old version
-# of these files can cause random problems/crashes, particularly
-# after an upgrade
-if [ -d $HOME/.kodi/temp ]; then
-  rm -f $HOME/.kodi/temp/*.fi
-fi
-
```
Comment (lrusak) [inline packages/mediacenter/kodi/scripts/kodi-config]: any reason for removing this?

File: tools/mkpkg/mkpkg_kodi [hunk 3]
```diff
@@ -31,7 +31,7 @@ DEST_DIR="$PKG_NAME-$BRANCH"
 
 echo "getting sources..."
   if [ ! -d $DEST_DIR.git ]; then
-    git clone --depth 1 $GIT_REPO -b $BRANCH $DEST_DIR.git
+    git clone --depth 128 $GIT_REPO -b $BRANCH $DEST_DIR.git
```
Comment (lrusak) [inline tools/mkpkg/mkpkg_kodi]: What's the reason for this?

File: packages/mediacenter/kodi-theme-Estuary/package.mk (hunk 1 above)
Comment (MilhouseVH) [inline packages/mediacenter/kodi-theme-Estuary/package.mk]: No, `icon.png` has moved to the resources folder after https://github.com/xbmc/xbmc/pull/10185

The `resources` folder is already copied in line 50.

File: packages/mediacenter/kodi/scripts/kodi-config (hunk 2 above)
Comment (MilhouseVH) [inline packages/mediacenter/kodi/scripts/kodi-config]: Yep, cache files are now stored in `.kodi/temp/archive_cache` (https://github.com/xbmc/xbmc/pull/9225).

Although there's a flaw with 9225 (see comments), which https://github.com/xbmc/xbmc/pull/10206 fixes. We could add 10206 to this PR if you like, either way the above code is no longer required.

**Comment by lrusak** [comment]
LGTM

File: tools/mkpkg/mkpkg_kodi (hunk 3 above)
Comment (MilhouseVH) [inline tools/mkpkg/mkpkg_kodi]: It's in the first comment.

By default `mkpkg_kodi master` will use the last commit (at the time, Beta1) when we actually wanted the commit before (Alpha3) in order to create the kodi-17-alpha3 package. The current `mkpkg_kodi` does take a second optional argument which is a specific rev, but with `--depth 1` it's not possible to select a different revision other than the last commit (Beta1).

**Comment by MilhouseVH** [comment]
Added https://github.com/xbmc/xbmc/pull/10206.

**Comment by MilhouseVH** [comment]
Note: Building RPi/RPi2 for current master will fail without this PR, as the latest RPi firmware (bumped in #481) uses alternate `libbrcmEGL`/`libbrcmGLESv2` libraries instead of `libEGL`/`libGLESv2`, and this new firmware will cause pre-alpha3 Kodi to fail to build. The Kodi support patches in this PR address this change.

**Pull request merged**